### PR TITLE
/etc/os-release generator

### LINKED ISSUE #165: implement /etc/os-release mutation
For scenarios where we are not pulling `alpine-base`, we need to generate our own `/etc/os-release` file.  We should probably also generate `/etc/<distro>-release` as well based on the `/etc/os-release` contents.

### PR DESCRIPTION
**Pull request by kaniini**
Generate `/etc/os-release` if it does not already exist.

Closes #165.

### CODE REVIEW

File: pkg/build/os-release.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			return err
+		}
+	}
+
+	if ic.OSRelease.BugReportURL != "" {
+		_, err := fmt.Fprintf(w, "BUG_REPORT_URL=\"%s\"\n", ic.OSRelease.BugReportURL)
+		if err != nil {
+			return err
+		}
+	}
```
Comment (puerco) [inline pkg/build/os-release.go:83]: What do you think if we do this in a template later to skip all the ifs and error checks?

**Review by puerco — APPROVED** [review]

File: pkg/build/os-release.go (hunk 1 above)
Comment (kaniini) [reply pkg/build/os-release.go:83]: sure

File: examples/alpine-slim.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+environment:
+  PATH: /usr/sbin:/sbin:/usr/bin:/bin
+
+# data for /etc/os-release if it does not already exist
+# in the image
+os-release:
+  id: alpine
+  version-id: '3.16.0'
+  name: 'Alpine Slim'
+  pretty-name: 'Alpine Slim (apko)'
```
Comment (mattmoor) [inline examples/alpine-slim.yaml:22]: Can we default this, or do we need to update all the distroless repos with this block?  I’d love for things to work with scanners by default 🤞

Comment (kaniini) [reply examples/alpine-slim.yaml:22]: it has defaults, but we need to figure out what sensible ones are

File: pkg/build/types/image_configuration.go [hunk 3]
```diff
@@ -64,6 +64,23 @@ func (ic *ImageConfiguration) Validate() error {
...
+		ic.OSRelease.ID = "alpine"
+	}
+
+	if ic.OSRelease.Name == "" {
+		ic.OSRelease.Name = "apko-generated image"
+		ic.OSRelease.PrettyName = "apko-generated image"
+	}
+
+	if ic.OSRelease.VersionID == "" {
+		ic.OSRelease.VersionID = "3.999"
```
Comment (mattmoor) [inline pkg/build/types/image_configuration.go:77]: I assume this is what you mean by sensible?

Comment (mattmoor) [reply pkg/build/types/image_configuration.go:77]: Or things we need to make sensible…? 😂

**Comment by mattmoor** [comment]
I wonder if we should add test files to each of the example directories and run something like:

   go test ./dir —digest=…

We can make them pretty trivial, but then we have a pattern for testing interesting properties of each example image.  In this case, the shape of /etc/os-release

File: pkg/build/types/image_configuration.go (hunk 3 above)
Comment (kaniini) [reply pkg/build/types/image_configuration.go:77]: Things we need to make sensible, I guess.  "edge" is probably a good versionid to use here.

**Comment by kaniini** [comment]
> I wonder if we should add test files to each of the example directories and run something like:
> 
> go test ./dir —digest=…
> 
> We can make them pretty trivial, but then we have a pattern for testing interesting properties of each example image. In this case, the shape of /etc/os-release

Lets focus on this for 0.4!

**Pull request merged**